faster_whisper/transcribe.py

```diff
@@ -504,16 +512,21 @@ def generate_segments(
                 all_tokens.extend(tokens)
 
                 yield Segment(
+                    id=idx,
```

Comment:
The id goes back to 0 when moving to the next 30-second window. You should increment a global index instead.